src/core/InitiativeOrder/InitiativeOrder.ts

```diff
@@ -0,0 +1,231 @@
+import {
+  InitiativeItem,
+  InitiativeOrderState
+} from '@core/types';
+
+export interface InitiativeOrderInterface {
+  getCurrentId(): string;
+  getItems(): InitiativeItem[];
+  getRound(): number;
+  hide(id: string): void;
+  next(): void;
+  prev(): void;
+  remove(id: string): void;
+  reset(): void;
+  reveal(id: string): void;
+  setCurrentId(id: string): void;
+  setItems(items: InitiativeItem[]): void;
+  setResourceA(id: string, value: number): void;
+  setResourceB(id: string, value: number): void;
+  setSortValue(id: string, value: number): void;
+  getState(): InitiativeOrderState;
+  setRound(round: number): void;
+  sort: () => void;
+}
+
+export class InitiativeOrder implements InitiativeOrderInterface {
+  private currentId: string;
+  private items: InitiativeItem[];
+  private round: number;
+  
+  constructor() {
+    this.currentId  =  '';
+    this.items = [];
+    this.round = 1;
+  }
+
+  getCurrentId(): string {
+    return this.currentId;
+  }
+
+  getItems(): InitiativeItem[] {
+    return this.items;
+  }
+
+  getRound(): number {
+    return this.round;
+  }
+
+  getState(): InitiativeOrderState {
+    return {
+      currentId: this.currentId,
+      items: this.items,
+      round: this.round
+    };
+  }
+
+  hide(id: string) {
+    const newItems = this.items.map((item) => {
+      if (item.id === id) {
+        item.visibilityState = 'hidden';
+      }
+
+      return item;
+    });
+
+    this.items = newItems;
+  }
+
+  next() {
+    const currentItemIndex = this.items.findIndex((item) => item.id === this.currentId); 
+    let nextItem: InitiativeItem | null = null;
+
+    let x = currentItemIndex;
+    while(!nextItem) {
+      const nextIndex = x === this.items.length - 1
+        ? 0
+        : x+1;
+      
+      const n = this.items[nextIndex];
+
+      if (
+        n.visibilityState === 'on' ||
+        n.visibilityState === 'hidden'
+      ) {
+        nextItem = n;
+      }
+
+      x = nextIndex;
+    }
+
+    if (currentItemIndex + 1 === this.items.length) {
+      this.round = this.round + 1;
+    }
+
+    this.currentId = nextItem?.id ?? '';
+  }
+
+  prev() {
+    const currentItemIndex = this.items.findIndex((item) => item.id === this.currentId); 
+    let prevItem: InitiativeItem | null = null;
+
+    let x = currentItemIndex;
+    while(!prevItem) {
+      const prevIndex = x === 0
+        ? this.items.length - 1
+        : x-1;
+      
+      const n = this.items[prevIndex];
+
+      if (
+        n.visibilityState === 'on' ||
+        n.visibilityState === 'hidden'
+      ) {
+        prevItem = n;
+      }
+
+      x = prevIndex;
+    }
+
+    if (currentItemIndex === 0 && this.round > 1) {
+      this.round = this.round - 1;
+    }
+
+    this.currentId = prevItem?.id ?? '';
+  }
+
+  remove = (id: string) => {
+    const filtered = this.items.map((item) => {
```

Comment:
nit: this is a nit I'd do at a job, but I think you can use the filter method instead of map to get the item with the matching ID

Reply:
Heh, this needs to be redone at some point and this is a good call out. I'm cheating here by keeping removed items around and depending on CSS animations to handle removal in a smooth way. Really there needs to be a step that does the animation before applying the new state, but that's a future refactor.

But also, there's valid uses for this kind of logic down the road, such as initiative cards that are only visible to the GM and not the player, so I'm keeping it around for now.